src/map/status.cpp

```diff
@@ -5841,13 +5841,28 @@ void status_calc_bl_main(struct block_list *bl, std::bitset<SCB_MAX> flag)
 	if(flag[SCB_SPEED]) {
 		status->speed = status_calc_speed(bl, sc, b_status->speed);
 
-		if( bl->type&BL_PC && !(sd && sd->state.permanent_speed) && status->speed < battle_config.max_walk_speed )
-			status->speed = battle_config.max_walk_speed;
+		if (bl->type & BL_PC && sd != nullptr) {
```

Comment:
```suggestion
		if (sd != nullptr) {
```
sd was already assigned with BL_CAST and checked the bl->type